### PR TITLE
docs: Nexus as Container Registry in KRCI

### PR DESCRIPTION
**Pull request by alezander86**
## Description

This documentation guide provides comprehensive instructions for integrating **Nexus Repository Manager** as a docker image registry with KubeRocketCI.

## Type of change

- [x] documentation

## Checklist

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Pull Request contain one commit. I squash my commits.

### CODE REVIEW

File: docs/operator-guide/artifacts-management/nexus-image-registry.md [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+      }
+      ```
+      </TabItem>
+  </Tabs>
+
+## Configuration
+
+With [Add-ons approach](https://github.com/epam/edp-cluster-add-ons/blob/main/clusters/core) `nexus-operator` create the necessary roles,**Service Account** `ci.user`, blob stores, repository, scripts, cleanup policies automatically. Below is a comprehensive guide on how to create all resources manually.
+
+## Configuration Nexus Repository Manager with [Add-ons approach](https://github.com/epam/edp-cluster-add-ons/blob/main/clusters/core)
```
Comment (Rolika4) [inline docs/operator-guide/artifacts-management/nexus-image-registry.md:70]: \### Configuration Nexus Repository Manager with [Add-ons approach]

Comment (Rolika4) [reply docs/operator-guide/artifacts-management/nexus-image-registry.md:70]: Do we use link at title?

File: docs/operator-guide/artifacts-management/nexus-image-registry.md [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+      ```json
+      "ci-nexus":
+      {
+        "password": "XXXXXXX"
+      }
+      ```
+      </TabItem>
+  </Tabs>
+
+## Configuration Nexus Repository Manager without [Add-ons approach](https://github.com/epam/edp-cluster-add-ons/blob/main/clusters/core)
```
Comment (Rolika4) [inline docs/operator-guide/artifacts-management/nexus-image-registry.md:102]: \### Configuration Nexus Repository Manager without [Add-ons approach]

File: docs/operator-guide/artifacts-management/nexus-image-registry.md [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+
+Without [Add-ons approach](https://github.com/epam/edp-cluster-add-ons/blob/main/clusters/core) you need to manually update: helm chart `Nexus Repository Manager` value and create blob stores, repository, roles, **Service Account** `ci.user`:
```
Comment (Rolika4) [inline docs/operator-guide/artifacts-management/nexus-image-registry.md:104]: For what this link?
This step about configuring without add-ons, why you add link for add-ons?

Comment (Rolika4) [reply docs/operator-guide/artifacts-management/nexus-image-registry.md:104]: Without add-ons, it is necessary to update and create the resources listed below:

(without 'you')

File: docs/operator-guide/artifacts-management/nexus-image-registry.md [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+          enabled: true
+          registries:
+            - host: nexus-ci-container.example.com
+              port: 5000
+    ```
+
+2. Create blob store. Open the Nexus UI and navigate to **Server administration and configuration** -> **Repository** -> **Blob Stores** and click **Create Blob Store**, fill the following fields and click **Save** button:
+
+    * **Type**: `File`;
+    * **Name**: `edp-yum`;
```
Comment (Rolika4) [inline docs/operator-guide/artifacts-management/nexus-image-registry.md:121]: yum for docker images???

File: docs/operator-guide/artifacts-management/nexus-image-registry.md [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+
+    ![Nexus create blob store](../../assets/operator-guide/artifacts-management/nexus-registry-create-roles.png "Nexus create blob store")
```
Comment (Rolika4) [inline docs/operator-guide/artifacts-management/nexus-image-registry.md:123]: Incorrect image (create role) instead create blob store

File: docs/operator-guide/artifacts-management/nexus-image-registry.md [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+
+3. Create repository. Open the Nexus UI and navigate to **Server administration and configuration** -> **Repository** -> **Repositories**, click **+ Create repository**, select **docker (hosted)**, fill the following fields and click **Save** button:
+
+    * **Name**: `A unique identifier for this repository`;
+    * **HTTP**: `5000`;
+    * **Enable Docker V1 API**: `check`;
+    * **Blob store name**: `edp-yum`;
```
Comment (Rolika4) [inline docs/operator-guide/artifacts-management/nexus-image-registry.md:130]: yum?

File: docs/operator-guide/artifacts-management/nexus-image-registry.md [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+
+4. Open the Nexus UI and navigate to **Server administration and configuration** -> **Security** -> **Roles**. Click the **Create Role** button, fill the following fields and click **Save** button:
+
+    * **Type**: `Nexus role`;
+    * **Role ID**: `edp-admin`;
+    * **Role Name**: `edp-admin`;
+    * **Role Descriptions**: `Read and write access to all repos and scripts`;
+    * **Applied Privileges**: `nx-apikey-all`, `nx-repository-view-*-*-add`, `nx-repository-view-*-*-browse`,`nx-repository-view-*-*-edit`, `nx-repository-view-*-*-read`, `nx-script-*-add`, `nx-script-*-delete`, `nx-script-*-read`, `nx-script-*-run`;
+
+    ![Roles settings](../../assets/operator-guide/artifacts-management/nexus-settings-user.png "Roles settings")
```
Comment (Rolika4) [inline docs/operator-guide/artifacts-management/nexus-image-registry.md:143]: Incorrect image

File: docs/operator-guide/artifacts-management/nexus-image-registry.md [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+    * **Blob store name**: `edp-yum`;
+    * **Strict Content Type Validation**: `check`;
+
+    ![Nexus create registry](../../assets/operator-guide/artifacts-management/nexus-registry-create-repository.png "Nexus create registry")
+
+4. Open the Nexus UI and navigate to **Server administration and configuration** -> **Security** -> **Roles**. Click the **Create Role** button, fill the following fields and click **Save** button:
+
+    * **Type**: `Nexus role`;
+    * **Role ID**: `edp-admin`;
+    * **Role Name**: `edp-admin`;
```
Comment (Rolika4) [inline docs/operator-guide/artifacts-management/nexus-image-registry.md:139]: Could we remove `edp` from naming pattern?

File: docs/operator-guide/artifacts-management/nexus-image-registry.md [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+
+6. Type the `ci.user` username, fill the following fields and click the **Create local user** button to create the ***Service Account**, fill the following fields and click **Save** button::
+
+    * **ID**: `ci.user`;
+    * **First name**: `ci.user`;
+    * **Last name**: `CI`;
+    * **Email**: `<nexus-user-id>@krci.com`;
+    * **Password**: `<nexus-user-password>`;
+    * **Status**: `Active`;
+    * **Roles**: `edp-admin`;
```
Comment (Rolika4) [inline docs/operator-guide/artifacts-management/nexus-image-registry.md:157]: Could we remove `edp` from naming pattern?

File: docs/operator-guide/artifacts-management/nexus-image-registry.md [hunk 10]
```diff
@@ -0,0 +1,183 @@
...
+        "user": "XXXXXXX",
+        "password": "XXXXXXX"
+      }
+      ```
+      </TabItem>
+  </Tabs>
+
+## Configuration
+
+With [Add-ons approach](https://github.com/epam/edp-cluster-add-ons/blob/main/clusters/core) `nexus-operator` create the necessary roles,**Service Account** `ci.user`, blob stores, repository, scripts, cleanup policies automatically. Below is a comprehensive guide on how to create all resources manually.
```
Comment (Rolika4) [inline docs/operator-guide/artifacts-management/nexus-image-registry.md:68]: Add-Ons approach

File: docs/operator-guide/artifacts-management/nexus-image-registry.md (hunk 6 above)
Comment (alezander86) [reply docs/operator-guide/artifacts-management/nexus-image-registry.md:130]: https://github.com/epam/edp-cluster-add-ons/blob/main/clusters/core/addons/nexus-operator/templates/nexus/repository/yum/edp-yum-blobstore.yaml

File: docs/operator-guide/artifacts-management/nexus-image-registry.md (hunk 4 above)
Comment (alezander86) [reply docs/operator-guide/artifacts-management/nexus-image-registry.md:121]: https://github.com/epam/edp-cluster-add-ons/blob/main/clusters/core/addons/nexus-operator/templates/nexus/repository/yum/edp-yum-blobstore.yaml

File: docs/operator-guide/artifacts-management/nexus-image-registry.md [hunk 11]
```diff
@@ -0,0 +1,183 @@
...
+
+# Nexus Image Registry
+
+This documentation guide provides comprehensive instructions for integrating **Nexus Repository Manager** as a docker image registry with KubeRocketCI.
+
+## Installation
+
+To install Nexus in your environment, it's recommended to use the resources provided in the [Cluster Add-Ons](https://github.com/epam/edp-cluster-add-ons) repository. Here are detailed instructions for installing and managing the [Add-ons Application](../add-ons-overview.md). This approach involves installing both the Nexus repository manager and the [nexus-operator](https://github.com/epam/edp-nexus-operator). Leveraging the Cluster Add-Ons simplifies the deployment and management process, providing a streamlined method to integrate Nexus into your infrastructure.
+
+1. Navigate to Cluster Add-Ons repository and align [nexus values.yaml](https://github.com/epam/edp-cluster-add-ons/blob/main/clusters/core/addons/nexus/values.yaml) and [nexus-operator values.yaml](https://github.com/epam/edp-cluster-add-ons/blob/main/clusters/core/addons/nexus-operator/values.yaml):
```
Comment (daniil-nedostup) [inline docs/operator-guide/artifacts-management/nexus-image-registry.md:12]: lets add "... files" at the end of sentence.

e.g. Navigate to Cluster Add-Ons repository and align the [nexus values.yaml](https://github.com/epam/edp-cluster-add-ons/blob/main/clusters/core/addons/nexus/values.yaml) and [nexus-operator values.yaml](https://github.com/epam/edp-cluster-add-ons/blob/main/clusters/core/addons/nexus-operator/values.yaml) files.

File: docs/operator-guide/artifacts-management/nexus-image-registry.md [hunk 12]
```diff
@@ -0,0 +1,183 @@
...
+
+    ![Nexus ArgoCD applications](../../assets/operator-guide/artifacts-management/nexus-registry-addons-deploy.png "Nexus ArgoCD applications")
+
+3. **Sign In** into **Nexus Repository Manager** using the default credentials **Username**: `admin` and **Password**: `admin123` and update the default password for better security:
+
+    ![Nexus ArgoCD applications](../../assets/operator-guide/artifacts-management/nexus-registry-change-password.png "Nexus ArgoCD applications")
+
+4. Update the configuration to enable **nexus** and **nexus-operator** in [Add-Ons](https://github.com/epam/edp-cluster-add-ons/blob/main/clusters/core/apps/values.yaml) repository:
+
+    ```yaml
```
Comment (daniil-nedostup) [inline docs/operator-guide/artifacts-management/nexus-image-registry.md:24]: lets mention path of values.yaml file in title:
    ```yaml title="clusters/core/apps/values.yaml"

File: docs/operator-guide/artifacts-management/nexus-image-registry.md [hunk 13]
```diff
@@ -0,0 +1,183 @@
...
+          enabled: true
+          registries:
+            - host: nexus-ci-container.example.com
+              port: 5000
+    ```
+
+2. Create blob store. Open the Nexus UI and navigate to **Server administration and configuration** -> **Repository** -> **Blob Stores** and click **Create Blob Store**, fill the following fields and click **Save** button:
+
+    * **Type**: `File`;
+    * **Name**: `edp-container`;
```
Comment (daniil-nedostup) [inline docs/operator-guide/artifacts-management/nexus-image-registry.md:121]: krci*

**Review by MykolaMarusenko — APPROVED** [review]

**Pull request merged**